### PR TITLE
Add coverage to tests, add display-name for nicer display on Charmhub

### PR DESCRIPTION
**Pull request by mthaddon**
The charmcraft init template now includes coverage, update this charm to include that.

Also add a "display-name" in metadata.yaml so the name of the charm is correctly displayed on https://charmhub.io

### CODE REVIEW

**Review by balbirthomas — DISMISSED** [review]
Thanks

File: run_tests [hunk 1]
```diff
@@ -13,4 +13,5 @@ else
 fi
 
 black --diff
-python3 -m unittest -v "$@"
+coverage run --branch --source=lib,src -m unittest -v "$@"
```
Comment (sed-i) [inline run_tests:16]:
```suggestion
coverage run --branch --source=src,lib/charms/prometheus_k8s -m unittest -v "$@"
```
Should coverage exclude external libs?

Comment (balbirthomas) [reply run_tests:16]: Sounds like a good idea. Particularly so, since unit tests for the charm library already exist :-) .

Comment (mthaddon) [reply run_tests:16]: I think Leon's suggestion makes sense though. `lib/charms/prometheus_k8s` ensures you're only running tests for your own libraries, not for others you may import in the future. I'll commit that.

File: .gitignore [hunk 2]
```diff
@@ -1,6 +1,7 @@
 *~
 *swp
 *.charm
+.coverage
```
Comment (dstathis) [inline .gitignore:4]: I think we want to have everyone using the same coverage settings so we should probably commit this file

**Review by dstathis — DISMISSED** [review]

File: .gitignore (hunk 2 above)
Comment (balbirthomas) [reply .gitignore:4]: This is an autogenerated coverage report. I do not see why it should be committed.

**Review by balbirthomas — APPROVED** [review]

**Review by balbirthomas — APPROVED** [review]

**Review by Abuelodelanada — APPROVED** [review]
Looks good to me

**Review by dstathis — APPROVED** [review]

File: .gitignore (hunk 2 above)
Comment (dstathis) [reply .gitignore:4]: Ah I see. I thought this was .coveragerc

It's a little bit weird they would start the report file with "."

**Pull request merged**